Lib/ufo2ft/kernFeatureWriter.py

```diff
@@ -87,33 +87,41 @@ def write(self, linesep="\n"):
             self._splitRtlKerning()
 
         # write the lookups and feature
+        ltrKern = []
         if self.ltrScripts or not self.rtlScripts:
+            self._addKerning(ltrKern, self.glyphPairKerning)
+            self._addKerning(ltrKern, self.leftClassKerning, enum=True)
+            self._addKerning(ltrKern, self.rightClassKerning, enum=True)
+            self._addKerning(ltrKern, self.classPairKerning, ignoreZero=True)
+        if ltrKern:
             lines.append("lookup kern_ltr {")
-            self._addKerning(lines, self.glyphPairKerning)
-            self._addKerning(lines, self.leftClassKerning, enum=True)
-            self._addKerning(lines, self.rightClassKerning, enum=True)
-            self._addKerning(lines, self.classPairKerning, ignoreZero=True)
+            lines.extend(ltrKern)
             lines.append("} kern_ltr;")
             lines.append("")
 
+        rtlKern = []
         if self.rtlScripts:
-            lines.append("lookup kern_rtl {")
-            self._addKerning(lines, self.rtlGlyphPairKerning, rtl=True)
-            self._addKerning(lines, self.rtlLeftClassKerning, rtl=True,
+            self._addKerning(rtlKern, self.rtlGlyphPairKerning, rtl=True)
+            self._addKerning(rtlKern, self.rtlLeftClassKerning, rtl=True,
                              enum=True)
-            self._addKerning(lines, self.rtlRightClassKerning, rtl=True,
+            self._addKerning(rtlKern, self.rtlRightClassKerning, rtl=True,
                              enum=True)
-            self._addKerning(lines, self.rtlClassPairKerning, rtl=True,
+            self._addKerning(rtlKern, self.rtlClassPairKerning, rtl=True,
                              ignoreZero=True)
+        if rtlKern:
+            lines.append("lookup kern_rtl {")
+            lines.extend(rtlKern)
             lines.append("} kern_rtl;")
             lines.append("")
 
         lines.append("feature kern {")
-        if self.ltrScripts or not self.rtlScripts:
+        if (self.ltrScripts or not self.rtlScripts) and ltrKern:
```

Comment:
OK right, this doesn't do what I intended. My goal was to make `kern_ltr` the default for all script/language pairs not mentioned in the feature source, and then make sure that all LTR scripts have the `kern_ltr` lookup and not the `kern_rtl` lookup while all RTL scripts have the opposite. I'm not sure how to do this with feature syntax. As it is, I think every script/language pair gets `kern_ltr` (and LRT scripts get it twice as you say).